From f16ffbf54ce49eb39b75b3d5f7e3402385dee18a Mon Sep 17 00:00:00 2001
From: Yann Sionneau <yann.sionneau@vates.tech>
Date: Thu, 21 May 2026 11:29:28 +0200
Subject: [PATCH] add debug messages for pci devices

---
 hw/pci/pci.c | 16 +++++++++++++---
 1 file changed, 13 insertions(+), 3 deletions(-)

diff --git a/hw/pci/pci.c b/hw/pci/pci.c
index 4a84e478ce..ceecaa3775 100644
--- a/hw/pci/pci.c
+++ b/hw/pci/pci.c
@@ -51,7 +51,7 @@
 #include "qapi/qapi-commands-pci.h"
 #include "qemu/cutils.h"
 
-//#define DEBUG_PCI
+#define DEBUG_PCI
 #ifdef DEBUG_PCI
 # define PCI_DPRINTF(format, ...)       printf(format, ## __VA_ARGS__)
 #else
@@ -2335,10 +2335,16 @@ static void pci_add_option_rom(PCIDevice *pdev, bool is_default_rom,
     char name[32];
     const VMStateDescription *vmsd;
 
-    if (!pdev->romfile)
+    PCI_DPRINTF("%s: device id: %04x\n", __func__, pdev->config[PCI_DEVICE_ID]);
+
+    if (!pdev->romfile) {
+	PCI_DPRINTF("%s: [dev id %04x] no romfile, returning\n", __func__, pdev->config[PCI_DEVICE_ID]);
         return;
-    if (strlen(pdev->romfile) == 0)
+    }
+    if (strlen(pdev->romfile) == 0) {
+	PCI_DPRINTF("%s: romfile length of 0, returning\n", __func__);
         return;
+    }
 
     if (!pdev->rom_bar) {
         /*
@@ -2352,14 +2358,17 @@ static void pci_add_option_rom(PCIDevice *pdev, bool is_default_rom,
          * if the rom bar is disabled.
          */
         if (DEVICE(pdev)->hotplugged) {
+	    PCI_DPRINTF("%s: hotplugged, returning\n", __func__);
             error_setg(errp, "Hot-plugged device without ROM bar"
                        " can't have an option ROM");
             return;
         }
 
         if (class == 0x0300) {
+	    PCI_DPRINTF("%s: calling rom_add_vga()\n", __func__);
             rom_add_vga(pdev->romfile);
         } else {
+	    PCI_DPRINTF("%s: calling rom_add_option()\n", __func__);
             rom_add_option(pdev->romfile, -1);
         }
         return;
@@ -2418,6 +2427,7 @@ static void pci_add_option_rom(PCIDevice *pdev, bool is_default_rom,
         pci_patch_ids(pdev, ptr, size);
     }
 
+    PCI_DPRINTF("%s: calling pci_register_bar()\n", __func__);
     pci_register_bar(pdev, PCI_ROM_SLOT, 0, &pdev->rom);
 }
 
-- 
2.53.0

