From 43ed96ad172c3e22105c7460f76f3761e5943e8d Mon Sep 17 00:00:00 2001
From: Yann Sionneau <yann.sionneau@vates.tech>
Date: Wed, 1 Jul 2026 11:25:23 +0200
Subject: [PATCH 3/3] pci: add debug in pci_default_write_config for EXP ROM
 BAR

---
 hw/pci/pci.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/hw/pci/pci.c b/hw/pci/pci.c
index d8812acdbe..48a31b5f41 100644
--- a/hw/pci/pci.c
+++ b/hw/pci/pci.c
@@ -1447,7 +1447,7 @@ void pci_default_write_config(PCIDevice *d, uint32_t addr, uint32_t val_in, int
     uint32_t val = val_in;
 
     assert(addr + l <= pci_config_size(d));
-
+ 
     for (i = 0; i < l; val >>= 8, ++i) {
         uint8_t wmask = d->wmask[addr + i];
         uint8_t w1cmask = d->w1cmask[addr + i];
@@ -1455,6 +1455,12 @@ void pci_default_write_config(PCIDevice *d, uint32_t addr, uint32_t val_in, int
         d->config[addr + i] = (d->config[addr + i] & ~wmask) | (val & wmask);
         d->config[addr + i] &= ~(val & w1cmask); /* W1C: Write 1 to Clear */
     }
+
+    if (addr == PCI_ROM_ADDRESS) {
+    	PCI_DPRINTF("%s: written to emulated PCI device d->config[0x30]: %04x\n", __func__, d->config[addr]);
+    }
+
+
     if (ranges_overlap(addr, l, PCI_BASE_ADDRESS_0, 24) ||
         ranges_overlap(addr, l, PCI_ROM_ADDRESS, 4) ||
         ranges_overlap(addr, l, PCI_ROM_ADDRESS1, 4) ||
-- 
2.53.0

